content/docs/intro/concepts/state.md

```diff
@@ -139,221 +98,167 @@ Enter your access token from https://app.pulumi.com/account/tokens
-$ pulumi stack select my-app-production
+It is possible to host your own version of the Pulumi Service in your private cloud environment. Pulumi offers versions that run natively on AWS, Azure, GCP, Kubernetes, or simple virtual machine-based private and hybrid cloud environments. The architecture is very similar, but is privately hosted and does not depend on public access over the Internet:
 
-# export the stack's checkpoint to a local file
-$ pulumi stack export --show-secrets --file my-app-production.checkpoint.json
-$ pulumi logout
-$ pulumi login
+<img src="/images/docs/reference/state_enterprise.png" alt="Pulumi Enterprise Architecture" class="img-bordered">
 
-# create a new stack with the same name on pulumi.com
-$ pulumi stack init my-app-production
+To learn more about self-host options, see [Self-Hosted Pulumi Service]({{< relref "/docs/guides/self-hosted" >}}) or [Contact Us]({{< relref "/pricing#contact" >}}).
 
-# import the new existing checkpoint into pulumi.com
-$ pulumi stack import --file my-app-production.checkpoint.json
-```
+## Advanced State
+
+Pulumi is designed to abstract state management away from you so that you can operate in terms of declarative infrastructure as code. In certain advanced cases, you may want or need to interact with state more directly. In those cases, the following topics may be helpful.
+
+> Due to the nature of the Pulumi Service backend being a managed service, it is the best at recovering from failure gracefully and shielding you from advanced state topics. If you choose to use an alternative backend, it is likely you'll need to be more familiar with the topics in this section.
+
+### Importing Existing Resources
+
+Pulumi supports importing resources that were already created resources outside of Pulumi, such as resources created using the cloud console, a cloud CLI or SDK, or even another infrastructure as code tool. The basic support imports resource metadata into your Pulumi state, and can also generate the source code in your chosen language to match that state.
+
+To learn more about importing existing resources, see [Importing Infrastructure]({{< relref "/docs/guides/adopting/import" >}}).
+
+### Checkpoints
+
+Pulumi state is usually stored in a transactional snapshot called a _checkpoint_. Pulumi records checkpoints early and often as it executes so that it can operate reliably, similar to how database transactions work. The basic functions of state allow Pulumi to diff your program's goal state against the last known update, recover from failure, and destroy resources accurately to clean up afterwards, and the checkpoint format augments this with additional failure recovery capabilities in the face of partial failure. The Pulumi Service backends records every checkpoint so that it is possible to recover from exotic failure scenarios that other backends may have more trouble recovering from as they typically store a singular Pulumi state file.
+
```

Comment:
```suggestion
Pulumi state is usually stored in a transactional snapshot called a _checkpoint_. Pulumi records checkpoints early and often as it executes so that Pulumi can operate reliably, similar to how database transactions work. The basic functions of state allow Pulumi to diff your program's goal state against the last known update, recover from failure, and destroy resources accurately to clean up afterwards. The checkpoint format augments this with additional failure recovery capabilities in the face of partial failure.

The Pulumi Service backends records every checkpoint so that it is possible to recover from exotic failure scenarios. Self-managed backends may have more trouble recovering from these situations as they typically store a singular Pulumi state file.

```